### DISCUSSION TITLE
Should Renovate use the GCP/AWS Metadata Service implicitely?

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#26107] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.56.3

### If you're self-hosting Renovate, select which platform you are using.

GitLab

### What is your question?

We noticed that Renovate tries to perform requests against the [GCP Metadata endpoints](https://cloud.google.com/compute/docs/storing-retrieving-metadata) while resolving Docker dependencies, because of this line:

https://github.com/renovatebot/renovate/blob/main/lib/modules/datasource/util.ts#L20

During analysis we also noticed that Renovate tries to use the AWS Metadata Service implicitly, probably because of this line:

https://github.com/search?q=repo%3Arenovatebot%2Frenovate%20fromNodeProviderChain&type=code

We do block this on our hosts to reduce the risk of exposing credentials accidentally.

Do you think it would make sense to make this configurable? IMO it's of similar criticality as `exposeAllEnv` or `detectHostRulesFromEnv` so it might even be a candidate for opt-in?
  
### Logs (if relevant)

Not relevant

**Comment by rarkins** [comment]
Could be. If so then it would be nice to have a way to block it centrally than per-datasource, in case there are more later. Does the aws-machine-image work without it? i.e. would this disable the datasource, or just be disabling implicit credentials and leaving e.g. the traditional env approach

**Reply by fgreinacher** [reply]
Agreed! It's probably not only relevant for the AMI datasource, but for all datasources that interact with AWS APIs.

Based on the concepts I know I would have expected AWS authentication to be done via host rules only (unless instructed otherwise).

**Reply by rarkins** [reply]
@fgreinacher could you join our Slack chat for Renovate to discuss?

**Reply by fgreinacher** [reply]
We found that this is actually the GCP metadata endpoint. I have updated the description.

**Comment by rarkins** [comment]
It sounds like a new option named something like `exposeCloudMetadata` could be added, defaulting to true. And then the relevant google and aws datasources should check for it before configuring/attempting

**Reply by fgreinacher** [reply, marked as answer]
This has been implemented in https://github.com/renovatebot/renovate/pull/26411
